ignite/engine/engine.py

```diff
@@ -333,23 +282,7 @@ def has_event_handler(self, handler: Callable, event_name: Optional[Any] = None)
             event_name: The event the handler attached to. Set this
                 to ``None`` to search all events.
         """
-        if event_name is not None:
-            if event_name not in self._event_handlers:
-                return False
-            events = [event_name]  # type: Union[List[Any], Dict[Any, List]]
-        else:
-            events = self._event_handlers
-        for e in events:
-            for h, _, _ in self._event_handlers[e]:
-                if self._compare_handlers(handler, h):
-                    return True
-        return False
-
-    @staticmethod
-    def _compare_handlers(user_handler: Callable, registered_handler: Callable) -> bool:
-        if hasattr(registered_handler, "_parent"):
-            registered_handler = registered_handler._parent()  # type: ignore[attr-defined]
-        return registered_handler == user_handler
+        return super(Engine, self).has_event_handler(handler, event_name=event_name)
```

Comment:
Finally, I think we can remove this methods and ensure that docs are taken from base class

Reply:
I have been trying to find a way to duplicate the docs from a class to another by the way we want and I didn't find anything, if we want to remove these methods from here, we can mention the list of methods for example under the `Engine` in `engine.rst`, something similar to what we did with `events`.
For example:
```
ignite.engine.engine
--------------------------

.. currentmodule:: ignite.engine.engine

.. autosummary::
    :nosignatures:
    :toctree: generated

    Engine
 
.. currentmodule:: ignite.base.event_driven

.. note::
    Engine class is now derived from :class:`ignite.base.event_driven`. 

# and we can list all methods that engine include either from engine of events_driven
```

Reply:
OK, let's keep things as they are for now